src/main/groovy/com/twcable/grabbit/jcr/ProtoNodeDecorator.groovy

```diff
@@ -97,6 +97,10 @@ class ProtoNodeDecorator {
                 log.debug "Added mixin ${value.stringValue} for : ${node.name}."
             }
             else {
+                JcrNodeDecorator decoratedNode = new JcrNodeDecorator(node)
+                if(decoratedNode.checkoutIfNecessary()){
+                    addMixins(property, node)
```

Comment:
To reiterate, this isn't just about writing mixins. If a node is checked in, we can't write/modify any property, or child node within or under that node. 

I think within JcrNodeDecorator, we need to wrap things like setProperty, and addMixin; intercept these calls; THEN perform checkout/checkin actions as needed